integration-tests/src/test/groovy/com/okta/sdk/tests/it/OAuth2IT.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.tests.it
+
+import com.okta.sdk.client.Client
+import com.okta.sdk.resource.application.*
+import com.okta.sdk.resource.user.User
+import com.okta.sdk.resource.user.UserBuilder
+import com.okta.sdk.resource.user.UserList
+import com.okta.sdk.tests.Scenario
+import com.okta.sdk.tests.it.util.ITSupport
+import org.testng.annotations.Test
+
+import static com.okta.sdk.tests.it.util.Util.assertPresent
+import static com.okta.sdk.tests.it.util.Util.validateUser
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.notNullValue
+
+/**
+ * Tests for OAuth2 flow
+ *
+ * Below env variables need to be set for OAUTH2 flow to be enabled:
+ *
+ * OKTA_CLIENT_ORGURL (org url)
+ * OKTA_CLIENT_AUTHORIZATIONMODE (should take the value "PrivateKey")
+ * OKTA_CLIENT_CLIENTID (client id)
+ * OKTA_CLIENT_SCOPES="okta.apps.read okta.apps.manage okta.users.read okta.users.manage" (e.g.)
+ * OKTA_CLIENT_PRIVATEKEY="~/.okta/privateKey.pem" (e.g.)
+ */
+class OAuth2IT extends ITSupport {
```

Comment:
We should probably chat with @vijetmahabaleshwar-okta on this one. It looks like this test would run fine when using SSWS tokens as well, which might give a false positive.

And in order to run the full ITs using `PrivateKey` you probably need a few more scopes. (and possibly unsettling the `OKTA_CLIENT_TOKEN` env)

Reply:
@bdemers the idea here is to ensure that the test does work for SSWS as well as OAuth2. In other words, the test shouldn't fail just because OAUTH2 env variables aren't set. For clarity, I had included comments at the top of the class.